File: test/language/templates/cpp98/InstantiateTypeAsStructFieldTest.cpp
#include "gtest/gtest.h"

#include "templates/instantiate_type_as_struct_field/InstantiateTypeAsStructField.h"

namespace templates
{
namespace instantiate_type_as_struct_field
{

TEST(InstantiateTypeAsStructFieldTest, readWrite)
{
    InstantiateTypeAsStructField instantiateTypeAsStructField;
    Test32& test32 = instantiateTypeAsStructField.getTest();
    test32.setValue(13);

    zserio::BitStreamWriter writer;
    instantiateTypeAsStructField.write(writer);
    size_t bufferSize = 0;
    const uint8_t* buffer = writer.getWriteBuffer(bufferSize);

    zserio::BitStreamReader reader(buffer, bufferSize);
    InstantiateTypeAsStructField readInstantiateTypeAsStructField(reader);

    ASSERT_TRUE(instantiateTypeAsStructField == readInstantiateTypeAsStructField);
}

} // namespace instantiate_type_as_struct_field
} // namespace templates

File: test/language/templates/java/templates/InstantiateTypeAsStructFieldTest.java
package templates;

import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;

import java.io.File;
import java.io.IOException;

import zserio.runtime.BitSizeOfCalculator;
import zserio.runtime.ZserioError;
import zserio.runtime.io.BitStreamReader;
import zserio.runtime.io.BitStreamWriter;
import zserio.runtime.io.FileBitStreamReader;
import zserio.runtime.io.FileBitStreamWriter;

import templates.instantiate_type_as_struct_field.InstantiateTypeAsStructField;
import templates.instantiate_type_as_struct_field.Test32;

public class InstantiateTypeAsStructFieldTest
{
    @Test
    public void readWrite() throws IOException
    {
        final InstantiateTypeAsStructField instantiateTypeAsStructField = new InstantiateTypeAsStructField();
        instantiateTypeAsStructField.setTest(new Test32(13));

        final BitStreamWriter writer = new FileBitStreamWriter(TEST_FILE);
        instantiateTypeAsStructField.write(writer);
        writer.close();
        final BitStreamReader reader = new FileBitStreamReader(TEST_FILE);

        InstantiateTypeAsStructField readInstantiateTypeAsStructField =
                new InstantiateTypeAsStructField(reader);
        reader.close();
        assertTrue(instantiateTypeAsStructField.equals(readInstantiateTypeAsStructField));
    }

    private static final File TEST_FILE = new File("test.bin");
}
